Homework13/PageObjects/WebTablePage.cs

```diff
@@ -0,0 +1,107 @@
+        private DemoQaWebElement _firstName1 = new(By.XPath("((//div[@role='rowgroup'])[1]//descendant::div[@role='gridcell'])[1]"));
+        private DemoQaWebElement _tableRows = new(By.XPath("//div[@class='rt-tbody']"));
+        private DemoQaWebElement _searchTextBox = new(By.XPath("//input[@id='searchBox']"));
+        private DemoQaWebElement _firstNameTextBox = new(By.XPath("//input[@id='firstName']"));
+        private DemoQaWebElement _lastNameTextBox = new(By.XPath("//input[@id='lastName']"));
+        private DemoQaWebElement _emailTextBox = new(By.XPath("//input[@id='userEmail']"));
+        private DemoQaWebElement _ageTextBox = new(By.XPath("//input[@id='age']"));
+        private DemoQaWebElement _salaryTextBox = new(By.XPath("//input[@id='salary']"));
+        private DemoQaWebElement _departmentTextBox = new(By.XPath("//input[@id='department']"));
+        private DemoQaWebElement _deleteButton = new(By.XPath("//span[@id='delete-record-1']"));
+        private DemoQaWebElement _closeButton = new(By.XPath("//span[contains(text(),'×')]"));
+
+        public bool IsFormValidated() => _form.GetClass().Contains("was-validated");
+
+        public void ClickAddNewEntryButton() => _addNewEntryButton.Click();
+
+        public void ClickCloseButton() => _closeButton.Click();
+
+        public void ClickSubmitButton() => _submitButton.Click();
+
+        public void ClickEditEntryButton(string firstName) => _editButton(firstName).Click();
+
+        public void ClickDeleteEntryButton() => _deleteButton.Click();
+
+        public string GetFirstName1Value() => _firstName1.Text;
+
+        public List<string> GetListOfNotEmptyRows()
+        {
+            var rows = _tableRows.FindElements(By.XPath("//div[@role='rowgroup']"));
+            List<string> rowsText = new List<string>();
+            for (int i = 0; i < rows.Count; i++)
+            {
+                var stringBuilder = new StringBuilder();
+                {
+                    IWebElement element = rows[i];
+                    string text = element.Text;
+                    stringBuilder.Append($"{text}");
+                }
+
+                var finalString = stringBuilder.ToString().Trim();
+                bool isNull = string.IsNullOrEmpty(finalString);
+                if (!isNull)
+                {
+                    rowsText.Add(finalString);
+                }
+
+            }
+
+            return rowsText;
+        }
+
+        public int GetCountOfRowsWithValue(string filterValue) => GetListOfNotEmptyRows().Count(x => x.Contains(filterValue));
+
+        public int GetCountOfRows(List<string> rows) => rows.Count;
+
+        public string GetTextOfElement(int index)
+        {
+            string text = GetRow(index).Text;
+            var actualString = text.Replace("\r\n", " ");
+
+            return actualString;
+        }
+
+        public void FillInSearchTextBox(string value) => _searchTextBox.SendKeys(value);
```

Comment:
All these actions are performed in popup. You need to create a separate page class for popup - use "popup" in naming. Method which perform opening the popup should return the type of this popup